src/test/testing_assert.cpp

```diff
@@ -64,22 +125,30 @@ bool check_table_equal(const std::shared_ptr<const Table>& opossum_table,
   auto opossum_matrix = _table_to_matrix(opossum_table);
   auto expected_matrix = _table_to_matrix(expected_table);
 
-  const auto generate_table_comparison = [&]() {
+  const auto generate_table_comparison = [&](const std::string& error_type, const std::string& error_msg,
+                                             const std::vector<std::pair<uint64_t, uint16_t>>& highlighted_cells = {}) {
     std::stringstream stream;
     stream << "========= Tables are not equal =========" << std::endl;
-    stream << _matrix_to_string("Actual Result", opossum_matrix);
-    stream << std::endl;
-    stream << _matrix_to_string("Expected Result", expected_matrix);
-    stream << "========================================" << std::endl;
+    stream << "------- Actual Result -------" << std::endl;
+    stream << _matrix_to_string(opossum_matrix, highlighted_cells, ANSI_COLOR_RED);
+    stream << "-----------------------------" << std::endl << std::endl;
+    stream << "------- Expected Result -------" << std::endl;
+    stream << _matrix_to_string(expected_matrix, highlighted_cells, ANSI_COLOR_GREEN);
+    stream << "-------------------------------" << std::endl;
+    stream << "========================================" << std::endl << std::endl;
+    stream << "Type of error: " << error_type << std::endl;
+    stream << error_msg << std::endl;
     return stream.str();
   };
 
   // compare schema of tables
   //  - column count
   if (opossum_table->column_count() != expected_table->column_count()) {
-    std::cout << generate_table_comparison() << "Number of columns is different. " << std::endl
-              << "Actual result: " << opossum_table->column_count()
-              << ", Expected result: " << expected_table->column_count();
+    const std::string error_type = "Column count mismatch";
+    const std::string error_msg = "Actual number of columns: " + std::to_string(opossum_table->column_count()) + "\n" +
+                                  "Expected number of columns: " + std::to_string(expected_table->column_count());
+
+    std::cout << generate_table_comparison(error_type, error_msg) << std::endl;
```

Comment:
Since `generate_table_comparison` is always immediately printed, why not move the printing part into lambda and rename it to `print_table_comparison`?

Reply:
Good idea :+1: